### PR TITLE
Fix linking to the requirements

### LINKED ISSUE #78: Doxygen links to the main README.md page not rendering on GitHub
Links to the ``Requirements`` section on the README.md (which is used as the mainpage in dox) do not render correctly on GitHub pages. They render correctly locally on my labtop. Need to investigate the issue, but it seems likely that there is a difference in the Doxygen version used. 

**Here how it looks on GitHub**
Screen Shot 2024-08-26 at 10 24 42 AM

**And here how it should render**
Screen Shot 2024-08-26 at 10 24 26 AM

### PR DESCRIPTION
**Pull request by oruebel**
Fix #78 

To avoid linking from the docs to the README.md (which is broken on GitHub Pages), I updated the docs to:

- Move code requirements to the user installation docs
- Move optional tool requirements for developers to the developer installation docs
- Create a separate mainpage for the Doxygen docs instead of using the README.md file. This keeps the GitHub README and docs separate
- Replace useage of @ syntax with \ syntax in Doxygen for consistency
- Add per-page table of contents for longer pages to ease navigation

### CODE REVIEW

**Comment by oruebel** [comment]
@stephprince please take a look

File: README.md [hunk 1]
```diff
@@ -20,25 +20,14 @@ Below is a high-level overview of the project structure and capabilities we are
...
-
 ## Installation
 
 See the [AqNWB Documentation](https://neurodatawithoutborders.github.io/aqnwb/) for installation and integration instructions. 
 * [User Installation](https://neurodatawithoutborders.github.io/aqnwb/install_page.html)
 * [Developer Installation](https://neurodatawithoutborders.github.io/aqnwb/dev_install_page.html)
+
+## Contributing and Legal
+
+For more information about license, contributing guidelines, code of conduct
```
Comment (stephprince) [inline README.md:31]:
```suggestion
For more information about the license, contributing guidelines, code of conduct
```

**Review by stephprince — APPROVED** [review]
Looks good! The table of contents are helpful.

**Comment by oruebel** [comment]
@stephprince could you re-approve. I committed the suggested change you added which dismissed your review.

**Review by stephprince — APPROVED** [review]

**Pull request merged**